master/internal/provisioner/provisioner.go

```diff
@@ -110,18 +111,14 @@ func (p *Provisioner) provision(ctx *actor.Context) {
 	}
 
 	ctx.Log().Debug("scale happening")
-	toTerminate := p.scaleDecider.findInstancesToTerminate(
-		ctx, p.provider.maxInstanceNum(),
-	)
-	if len(toTerminate) > 0 {
-		p.provider.terminate(ctx, toTerminate)
+	if toTerminate := p.scaleDecider.findInstancesToTerminate(); len(toTerminate.InstanceIDs) > 0 {
+		ctx.Log().Infof("terminating %d instances: %s",
```

Comment:
These logs lines are not necessarily correct. If you want to keep them, I would change it to `telling provider to terminate X instances`

Reply:
it might be confusing to mention "provider" since it's more of an internal concept. how about "decided to terminate X instances"

Reply:
Yeah that sounds good to me